### PR TITLE
add qos and priority class defination for ensurance

### PR DESCRIPTION
**Pull request by yan234280533**

### CODE REVIEW

File: ensurance/v1alpha1/types.go [hunk 1]
```diff
@@ -31,14 +134,11 @@ type QualityProbe struct {
...
 	HTTPGet *corev1.HTTPGetAction `json:"httpGet,omitempty"`
-	// Init delay time for handler
-	// Defaults to 5
-	// +optional
-	InitialDelaySeconds *int32 `json:"initialDelaySeconds,omitempty"`
-	// Timeout for request.
-	// Defaults to 0, instead not timeout
+
+	// TimeoutSeconds is the timeout for request
+	// Defaults to 60, instead not timeout
```
Comment (mfanjie) [inline ensurance/v1alpha1/types.go:142]: I did not understand `instead not timeout`

File: ensurance/v1alpha1/types.go [hunk 2]
```diff
@@ -15,6 +15,109 @@ const (
...
 )
 
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:resource:scope=Cluster,shortName=pl
+// +kubebuilder:subresource:status
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PriorityLevel defines the qos behaviours for this priority's pods
+type PriorityLevel struct {
```
[the remark this answers is not in the document]
Comment (yan234280533) [reply ensurance/v1alpha1/types.go]: done~

File: ensurance/v1alpha1/types.go [hunk 3]
```diff
@@ -15,6 +15,109 @@ const (
...
+	// +optional
+	ResourceRequestMutations []ResourceRequestMutation `json:"resourceRequestMutations,omitempty"`
+}
+
+// PriorityLevelStatus defines the desired status of PriorityLevel
+type PriorityLevelStatus struct {
+}
+
+type ResourcePriority struct {
+	CPUPriority       CPUPriority       `json:"cpuPriority,omitempty"`
```
[the remark this answers is not in the document]
Comment (yan234280533) [reply ensurance/v1alpha1/types.go]: done~

File: ensurance/v1alpha1/types.go (hunk 1 above)
Comment (yan234280533) [reply ensurance/v1alpha1/types.go]: done~

File: ensurance/v1alpha1/types.go [hunk 4]
```diff
@@ -15,6 +15,109 @@ const (
...
+	// +optional
+	ResourcePriority ResourcePriority `json:"resourcePriority,omitempty"`
+
+	// AvoidanceStrategy define the avoidance strategy for pods
+	// +optional
+	AvoidanceStrategy AvoidanceStrategy `json:"avoidanceStrategy,omitempty"`
+
+	// ResourceRequestMutations define if the service need to mutate resource to expand resource
+	// +optional
+	ResourceRequestMutations []ResourceRequestMutation `json:"resourceRequestMutations,omitempty"`
```
[the remark this answers is not in the document]
Comment (yan234280533) [reply ensurance/v1alpha1/types.go]: done～

**Pull request merged**